server/sandbox_run.go

```diff
@@ -84,16 +89,38 @@ func (s *Server) RunPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 
 	id, name, err := s.generatePodIDandName(kubeName, namespace, attempt)
 	if err != nil {
-		return nil, err
+		if strings.Contains(err.Error(), "already reserved for pod") {
```

Comment:
We could probably introduce a custom error here. (could be handled in follow-on).